src/apollo-client/index.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import {
+  ApolloClient,
+  gql,
+  InMemoryCache,
+  ApolloProvider,
+  useQuery,
+  makeVar,
+} from '@apollo/client';
+
+import {
+  reducer,
+  initialState,
+  selectCount,
+  incrementAction,
+  doubleAction,
+  createApp,
+} from '../common';
+
+const COUNT_QUERY = gql`
+  query CountQuery {
+    count @client
+  }
+`;
+
+const currentState = makeVar(initialState);
+
+const client = new ApolloClient({
+  cache: new InMemoryCache({
+    typePolicies: {
+      Query: {
+        fields: {
+          count() {
+            return currentState().count;
+          }
+        }
+      }
+    }
+  })
+});
+
+const useCount = () => {
+  const { loading, error, data } = useQuery(COUNT_QUERY);
+  return (!loading && !error && data) ? selectCount(data) : 0;
+};
+
+const useIncrement = () => {
+  return () => {
+    return currentState(reducer(currentState(), incrementAction));
+  };
+};
+
+const useDouble = () => {
+  return () => {
+    return currentState(reducer(currentState(), doubleAction));
+  };
+};
```

Comment:
Please note that the `useIncrement` and `useDouble` naming is a bit misleading here since these are not hooks, but I kept the naming as is to line up with what `createApp` is expecting.